src/pci.rs

```diff
@@ -116,6 +140,7 @@ struct PciDevice {
     bars: [PciBar; 6],
     vendor_id: u16,
     device_id: u16,
+    caps: [VirtioPciCap; PciCapabilityType::MaxCaps as usize - 1], // caps start at 1
```

Comment:
If you're not planning to support real PCI devices, this shortcut is fine. But if we care about having a clear separation between the PCI and virtio-pci bits, we should not have this field as part of the `PciDevice`.

Reply:
Yeh my plan is to split that into the new VirtioPCITransport which implements the VirtioTransport needed for the virtio implementation.